File: src/main/java/com/tesseractservers/Comparators.java
package com.tesseractservers;

import java.util.Comparator;

/**
 * Provides utility for {@link Comparator}s.
 * @since Core 1.0
 * @author Tassilo
 */
public class Comparators {

	/**
	 * Negates/Inverts the given comparator.
	 * @param source the comparator to invert
	 * @param <T> the type
	 * @return a comparator that behaves the opposite way
	 */
	public static <T> Comparator<T> negate(Comparator<T> source) {
		return (e1, e2) -> -source.compare(e1, e2);
	}



	/**
	 * Static class.
	 */
	private Comparators() {
	}

}

File: src/main/java/net/proxzi/Comparators.kt
package net.proxzi

/**
 * Provides utility for [Comparator]s.
 * @since Core 1.1
 * @author Tassilo
 */
object Comparators {

	/**
	 * Negates/Inverts the comparator.
	 * @param comparator the comparator to negate
	 * @param <T> the type
	 * @return a comparator that behaves the opposite way
	 */
	fun <T> negate(comparator: Comparator<T>): Comparator<T> {
		return Comparator { o1, o2 -> -comparator.compare(o1, o2) }
	}

}
